### PR TITLE
[DOCS] Reformat index stats API docs

### PR DESCRIPTION
**Pull request by jrodewig**
Updates the index stats API docs to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Also moves several shared parameters with the [Nodes Stats API](https://www.elastic.co/guide/en/elasticsearch/reference/master/cluster-nodes-stats.html) to the `common-parms` file.

Relates to elastic/docs#937 and #43765

### CODE REVIEW

File: docs/reference/rest-api/common-parms.asciidoc [hunk 1]
```diff
@@ -141,6 +184,74 @@ tag::index[]
...
+tag::index-metric[]
+`<index-metric>`::
++
+--
+(Optional, string)
+Comma-separated list of metrics used to limit the request.
+Supported metrics are:
+
+`_all`::
+Return all statistics
```
Comment (szabosteve) [inline docs/reference/rest-api/common-parms.asciidoc]:
```suggestion
Return all statistics.
```

File: docs/reference/rest-api/common-parms.asciidoc [hunk 2]
```diff
@@ -141,6 +184,74 @@ tag::index[]
...
+`search`::
+Search statistics including suggest statistics.
+You can include statistics for custom groups
+by adding an extra `groups` parameter
+(search operations can be associated with one or more groups).
+The `groups` parameter accepts a comma separated list of group names.
+Use `_all` to return statistics for all groups.
+
+`segments`::
+Retrieve the memory use of the open segments. Optionally, setting the
```
Comment (szabosteve) [inline docs/reference/rest-api/common-parms.asciidoc]: It seems that this sentence wasn't finished.

**Review by szabosteve — CHANGES_REQUESTED** [review]
I think there is an uncompleted sentence in the documentation.

File: docs/reference/rest-api/common-parms.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/rest-api/common-parms.asciidoc]: Thanks for catching that. Fixed with 2e4d91ef4c1ec6fb4122949d2e618ca7f5e6992f and 
730f68f.

**Review by szabosteve — APPROVED** [review]
LGTM

**Pull request merged**